sdk/include/platform/sunburst/platform-ethernet.hh

```diff
@@ -0,0 +1,633 @@
+	  RxBroadcastFrame          = 1 << 7,
+	  RxUdpFrameChecksumStatus  = 1 << 10,
+	  RxTcpFrameChecksumStatus  = 1 << 11,
+	  RxIpFrameChecksumStatus   = 1 << 12,
+	  RxIcmpFrameChecksumStatus = 1 << 13,
+	  RxFrameValid              = 1 << 15,
+	};
+
+	enum [[clang::flag_enum]] RxQueueControl : uint16_t{
+	  ReleaseRxErrorFrame            = 1 << 0,
+	  StartDmaAccess                 = 1 << 3,
+	  AutoDequeueRxQFrameEnable      = 1 << 4,
+	  RxFrameCountThresholdEnable    = 1 << 5,
+	  RxDataByteCountThresholdEnable = 1 << 6,
+	  RxDurationTimerThresholdEnable = 1 << 7,
+	  RxIpHeaderTwoByteOffsetEnable  = 1 << 9,
+	  RxFrameCountThresholdStatus    = 1 << 10,
+	  RxDataByteCountThresholdstatus = 1 << 11,
+	  RxDurationTimerThresholdStatus = 1 << 12,
+	};
+
+	enum [[clang::flag_enum]] TxQueueControl : uint16_t{
+	  ManualEnqueueTxQFrameEnable = 1 << 0,
+	  TxQMemoryAvailableMonitor   = 1 << 1,
+	  AutoEnqueueTxQFrameEnable   = 1 << 2,
+	};
+
+	enum [[clang::flag_enum]] Interrupt : uint16_t{
+	  EnergyDetectInterrupt             = 1 << 2,
+	  LinkupDetectInterrupt             = 1 << 3,
+	  ReceiveMagicPacketDetectInterrupt = 1 << 4,
+	  ReceiveWakeupFrameDetectInterrupt = 1 << 5,
+	  TransmitSpaceAvailableInterrupt   = 1 << 6,
+	  ReceiveProcessStoppedInterrupt    = 1 << 7,
+	  TransmitProcessStoppedInterrupt   = 1 << 8,
+	  ReceiveOverrunInterrupt           = 1 << 11,
+	  ReceiveInterrupt                  = 1 << 13,
+	  TransmitInterrupt                 = 1 << 14,
+	  LinkChangeInterruptStatus         = 1 << 15,
+	};
+
+	enum [[clang::flag_enum]] Port1Status : uint16_t{
+	  Partner10BTHalfDuplexCapability  = 1 << 0,
+	  Partner10BTFullDuplexCapability  = 1 << 1,
+	  Partner100BTHalfDuplexCapability = 1 << 2,
+	  Partner100BTFullDuplexCapability = 1 << 3,
+	  PartnerFlowControlCapability     = 1 << 4,
+	  LinkGood                         = 1 << 5,
+	  AutoNegotiationDone              = 1 << 6,
+	  MDIXStatus                       = 1 << 7,
+	  OperationDuplex                  = 1 << 9,
+	  OperationSpeed                   = 1 << 10,
+	  PolarityReverse                  = 1 << 13,
+	  HPMDIX                           = 1 << 15,
+	};
+
+	/**
+	 * The futex used to wait for interrupts when packets are available to
+	 * receive.
+	 */
+	const uint32_t *receiveInterruptFutex;
+
+	inline void set_gpio_output_bit(uint32_t bit, bool value) const
+	{
+		uint32_t output = gpio()->output;
+		output &= ~(1 << bit);
+		output |= value << bit;
+		gpio()->output = output;
+	}
+
+	/**
+	 * Read a register from the KSZ8851.
+	 */
+	uint16_t register_read(RegisterOffset reg) const
+	{
+		uint8_t addr = static_cast<uint8_t>(reg);
+		uint8_t be   = (addr & 0x2) == 0 ? 0b0011 : 0b1100;
+		uint8_t bytes[2];
+		bytes[0] = (0b00 << 6) | (be << 2) | (addr >> 6);
+		bytes[1] = (addr << 2) & 0b11110000;
+
+		set_gpio_output_bit(EthCsPin, false);
+		spi()->blocking_write(bytes, 2);
+		uint16_t val;
+		spi()->blocking_read((uint8_t *)&val, 2);
```

Comment:
The SPI protocol needs 2 bytes, hence `uint16_t`. Using `sizeof` makes it feel like that the size is arbitrary and it only needs to match the size of integer (at least for me)

Reply:
Same as above.  I believe that (from the SPI protocol) the size of the read and write must match, and so this would also benefit from a `static_assert(sizeof(bytes) == sizeof(val))`.

Reply:
That's incorrect.

Reply:
Then please explain (in comments) what the code is doing so that I don't have to guess.